### PR TITLE
#38903: move softmax and batch norm to device 2.0 api

### PR DESCRIPTION
**Pull request by bbradelTT**
### Ticket
Link to Github Issue #38903

### Problem description
There's a new device api

### What's changed
Use the new device api in softmax and batch norm

### Checklist

- [ ] [All post-commit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml?query=branch:bbradel-38903_fusedm20)
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:bbradel-38903_fusedm20)
- [ ] [cpp-unit-tests](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:bbradel-38903_fusedm20)
- [ ] New/Existing tests provide coverage for changes
- [ ] (Optional) Ran [clang-tidy code analysis](https://github.com/tenstorrent/tt-metal/actions/workflows/code-analysis.yaml) on the PR branch to catch linting errors early

#### Model tests

If your changes cover model-related code, you should run tests corresponding to affected models and platforms (Single card, T3K, Galaxy). "Choose your pipeline" workflows facilitate running multiple kinds of tests in a single run. Each offers `models-mandatory` and `models-extended` presets.
The former includes a minimal set of tests, to be run always. The latter extends that with additional ones - use your best judgement in deciding which is the most appropriate for your PR.

- [ ] [(Single) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:bbradel-38903_fusedm20)
  - [ ] `models-mandatory` preset (runs: [Device perf regressions](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) and [Frequent model and ttnn tests](https://github.com/tenstorrent/tt-metal/actions/workflows/fast-dispatch-full-regressions-and-models.yaml))
  - [ ] `models-extended` preset (runs: the mandatory tests, plus [Demo](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) and [Model perf](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) tests)
  - [ ] other selection - specify runs

- [ ] [(T3K) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:bbradel-38903_fusedm20)
  - [ ] `models-mandatory` preset (runs: [Unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml))
  - [ ] `models-extended` preset (runs: the mandatory tests, plus [Demo](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) and [Model perf](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-model-perf-tests.yaml) tests)
  - [ ] other selection - specify runs

- [ ] [(Galaxy) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:bbradel-38903_fusedm20)
  - [ ] `models-mandatory` preset (runs: [Quick tests](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml))
  - [ ] `models-extended` preset (runs: the mandatory tests, plus [Demo](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) and [Model perf](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-perf-tests.yaml) tests)
  - [ ] other selection - specify runs

### CODE REVIEW

**Review by nsorabaTT — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/batch_norm_sfpu_kernel.cpp [hunk 1]
```diff
@@ -31,9 +33,21 @@ ALWI void batchnorm_bcast_tiles(
     auto cb_affine_or_out = (weight_has_value || bias_has_value) ? cb_tmp_1 : cb_output_0;
     auto cb_scaled_output = (bias_has_value) ? cb_tmp_1 : cb_output_0;
 
+    experimental::CircularBuffer cb_bcast_obj(cb_bcast);
+    experimental::CircularBuffer cb_other_obj(cb_other);
+    experimental::CircularBuffer cb_batch_var_obj(cb_batch_var);
+    experimental::CircularBuffer cb_eps_obj(cb_eps);
```
[the remark this answers is not in the document]
Comment (bbradelTT) [reply ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/batch_norm_sfpu_kernel.cpp]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/tenstorrent/tt-metal/pull/39532#discussion_r2911980921)

[the remark this answers is not in the document]
Comment (bbradelTT) [reply ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/batch_norm_sfpu_kernel.cpp]: @copilot I will fix this in this PR. Please do not open a new one.

File: ttnn/cpp/ttnn/operations/normalization/softmax/device/kernels/attention/compute/softmax_large_tensor.cpp [hunk 2]
```diff
@@ -341,13 +359,25 @@ void kernel_main() {
...
+    experimental::CircularBuffer cb_fused_attn_obj(cb_fused_attn);
+    experimental::CircularBuffer cb_exps_obj(cb_exps);
+    experimental::CircularBuffer cb_scale_mask_obj(cb_scale_mask);
+    experimental::CircularBuffer cb_prev_reduce_obj(cb_prev_reduce);
+    experimental::CircularBuffer cb_in0_obj(cb_in0);
+    experimental::CircularBuffer cb_out0_obj(cb_out0);
+    experimental::CircularBuffer cb_x_obj(cb_x);
+    experimental::CircularBuffer cb_recip_obj(cb_recip);
+    experimental::CircularBuffer cb_prev_max_obj(cb_prev_max);
+    experimental::CircularBuffer cb_mask_padded_obj(cb_mask_padded);
```
[the remark this answers is not in the document]
Comment (bbradelTT) [reply ttnn/cpp/ttnn/operations/normalization/softmax/device/kernels/attention/compute/softmax_large_tensor.cpp]: Updated.

File: ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/batch_norm_kernel.cpp [hunk 3]
```diff
@@ -28,9 +30,21 @@ ALWI void batchnorm_bcast_tiles(
     auto cb_affine_or_out = (weight_has_value || bias_has_value) ? cb_tmp_1 : cb_output_0;
     auto cb_scaled_output = (bias_has_value) ? cb_tmp_1 : cb_output_0;
 
+    experimental::CircularBuffer cb_bcast_obj(cb_bcast);
+    experimental::CircularBuffer cb_other_obj(cb_other);
+    experimental::CircularBuffer cb_batch_var_obj(cb_batch_var);
+    experimental::CircularBuffer cb_eps_obj(cb_eps);
```
[the remark this answers is not in the document]
Comment (bbradelTT) [reply ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/batch_norm_kernel.cpp]: Updated

File: ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/batch_norm_sfpu_kernel.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (bbradelTT) [reply ttnn/cpp/ttnn/operations/normalization/batch_norm/device/kernels/compute/batch_norm_sfpu_kernel.cpp]: Updated
